cmd/talosctl/pkg/talos/action/node.go

```diff
@@ -0,0 +1,321 @@
+		// retryable function
+		err := func() error {
+			eventCh := make(chan client.EventResult)
+			err := a.cli.EventsWatchV2(a.ctx, eventCh, client.WithTailEvents(tailEvents))
+			if err != nil {
+				return err
+			}
+
+			if waitForActorID {
+				a.update(reporter.Update{
+					Message: "Waiting for actor ID",
+					Status:  reporter.StatusRunning,
+				})
+
+				select {
+				case actorID = <-actorIDCh:
+				case <-a.ctx.Done():
+					return a.ctx.Err()
+				}
+
+				a.update(reporter.Update{
+					Message: fmt.Sprintf("Actor ID: %v", actorID),
+					Status:  reporter.StatusRunning,
+				})
+
+				waitForActorID = false
+			}
+
+			return a.handleEvents(eventCh, actorID)
+		}()
+
+		// handle retryable errors
+
+		if errors.Is(err, context.Canceled) {
+			return nil
+		}
+
+		statusCode := client.StatusCode(err)
+		if statusCode == codes.Unavailable {
+			a.update(reporter.Update{
+				Message: "Unavailable, retrying...",
+				Status:  reporter.StatusError,
+			})
+
+			tailEvents = -1
+			actorID = ""
+
+			return retry.ExpectedError(err)
+		}
+
+		return err
+	})
+}
+
+func (a *nodeTracker) runPostCheckWithRetry() error {
+	return retry.Constant(a.tracker.retryDuration).Retry(func() error {
+		// retryable function
+		err := func() error {
+			err := a.tracker.postCheckFn(a.ctx, a.cli)
+			if err != nil {
+				return err
+			}
+
+			a.update(reporter.Update{
+				Message: "Post check passed",
+				Status:  reporter.StatusSucceeded,
+			})
+
+			return nil
+		}()
+
+		// handle retryable errors
+
+		if errors.Is(err, context.Canceled) {
+			return nil
+		}
+
+		statusCode := client.StatusCode(err)
+		if statusCode == codes.Unavailable {
+			a.update(reporter.Update{
+				Message: "Unavailable, retrying...",
+				Status:  reporter.StatusError,
+			})
+
+			return retry.ExpectedError(err)
+		}
+
+		return err
+	})
+}
+
+func (a *nodeTracker) handleEvents(eventCh chan client.EventResult, actorID string) error {
+	for {
+		var eventResult client.EventResult
+
+		select {
+		case eventResult = <-eventCh:
+		case <-a.ctx.Done():
+			return a.ctx.Err()
+		}
+
+		if a.tracker.expectedEventFn(eventResult) {
+			status := reporter.StatusSucceeded
+			if a.tracker.postCheckFn != nil {
+				status = reporter.StatusRunning
+			}
+
+			a.update(reporter.Update{
+				Message: "Events check condition met",
+				Status:  status,
+			})
+
+			return nil
+		}
+
+		if eventResult.Error != nil {
+			a.update(reporter.Update{
+				Message: fmt.Sprintf("Error: %v", eventResult.Error),
+				Status:  reporter.StatusError,
+			})
+
+			return eventResult.Error
+		}
+
+		if eventResult.Event.ActorID == actorID {
+			err := a.handleEvent(eventResult.Event)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (a *nodeTracker) handleEvent(event client.Event) error {
+	switch msg := event.Payload.(type) {
+	case *machineapi.PhaseEvent:
+		a.update(reporter.Update{
+			Message: fmt.Sprintf("PhaseEvent phase: %s action: %v", msg.GetPhase(), msg.GetAction()),
```

Comment:
nit, but we try to make our console messages lowercase, probably we could drop completely event name here and in other places?

```suggestion
			Message: fmt.Sprintf("phase: %s action: %v", msg.GetPhase(), msg.GetAction()),
```

Reply:
done